packages/grafana-ui/src/components/index.ts

```diff
@@ -41,7 +41,10 @@ export { ConfirmModal, ConfirmModalProps } from './ConfirmModal/ConfirmModal';
 export { QueryField } from './QueryField/QueryField';
 
 // Code editor
-export { CodeEditor } from './Monaco/CodeEditorLazy';
+export { CodeEditor } from './Monaco/CodeEditor';
+
+export { ReactMonacoEditorLazy as ReactMonacoEditor } from './Monaco/ReactMonacoEditorLazy';
+
```

Comment:
@gabor I see your point here with the CodeEditor API not being the same as Monaco's component API. But at the same time I don't like the idea of exporting pretty much the same components from grafana/ui just to expand the API surface. 

If the API is not sufficient for your needs (and that's totally fine, CodeEditor should still be considered experimental) I would suggest making CodeEditor props to implement the native API of Monaco, meaning extending the `EditorProps` interface rather than expanding grafana/ui API with yet another component. 

I believe our long term goal would be rather to limit the API surface of grafana/ui rather than expanding it with yet another component.

@mckn @jackw folks if you could loop in and share your thoughts on grafana/* packages API surface plans for the future, it would be quite valuable.

Reply:
@dprokop i also do not like the idea of re-exporting it, it's just the least-worst idea i have 😄 . the basic problem is this:

monaco is complicated to load in in the grafana project.

in a better world, i would just add monaco+react_monaco to the dependencies of grafana-main, and all would be good. but, because of lazy-loading, then i would have two copies of lazy-loaded-react-monaco (one made in grafana-ui, and one made by me), which i do not like.

extending the api of CodeEditor could work in this specific case, but i don't think long-term this approach is maintainable. you see, i already extended it once, `CodeEditor.onBeforeEditorDidMount` was done by me because i need it for Monaco. 

i will think some more about this.

Reply:
> extending the api of CodeEditor could work in this specific case, but i don't think long-term this approach is maintainable. you see, i already extended it once, CodeEditor.onBeforeEditorDidMount was done by me because i need it for Monaco.

Maybe I put my thoughts not super clear. I'm not suggesting to extend the CodeEditor prop by prop as the need occurs. I would rather change the API of CodeEditor all at once to be just EditorProps- exactly as you did in the newly ecported component.